### DISCUSSION TITLE
Batch userid parameter for getShopByOwnerUserId endpoint

### DISCUSSION THREAD

**Discussion by switmac** [etsy/open-api#662] [Ideas]
hi @etsyachristensen, we'd like to share an idea to
batch `userids` parameter for `getShopByOwnerUserId` endpoint
https://developer.etsy.com/documentation/reference#operation/getShopByOwnerUserId

in V2 we had a way to get these info without much api call by using the public endpoint
https://openapi.etsy.com/v2/users/$userids?includes=Shops,Profile

however in v3 
the endpoint https://openapi.etsy.com/v3/application/users/{user_id}/shops
allows only to get shops of each user one at a time.

**Comment by benjymouse** [comment]
+1

**Comment by etsyachristensen** [comment]
@switmac I think this request was made once before by someone. Unfortunately it was declined. We've gone through a lot of effort recently to ensure that the API isn't getting abused by outside developers (not that you are one, just that there has been a significant amount of abuse over the years). An endpoint like you are requesting makes it much easier for apps to harvest data from the API. The API was built for the intention of supplying our sellers with means to do some critical business processes that the Etsy system doesn't provide already. These endpoints are tools that are meant to be used to augment seller activities within their own shops. However, there have been a rash of apps that took advantage of the endpoints (especially in V2) open-nature and harvested the data to expose user data & sales numbers as well as market their tools for other nefarious purposes. So with V3 we are being very careful about these types of exposures. We do realize that having to make more API calls is certainly less ideal but it's a measure that has been intentionally taken for security.

But this board is here for a reason. We want to have discussion with developers around needs like these to find a way to more readily accommodate the requests while still maintaining our security. Very few things are 100% non-starters so please do share if you have some thoughts.

**Comment by emergingdzns** [comment]
What would be nice is if Etsy could provide a way in the [getShop](https://developers.etsy.com/documentation/reference#operation/getShop) endpoint (or add a `getShops` endpoint) to pass in a comma delimited list of shop ids so we can get back the shops with only their public data.
